app/controllers/api/reservations_controller.rb

```diff
@@ -0,0 +1,51 @@
+module Api
+  class ReservationsController < ApplicationController
+    before_action :set_reservation, only: %i[show update destroy]
+
+    def index
+      @reservations = Reservation.all.includes(:game)
+      render json: @reservations.as_json(include: :game)
+    end
+
+    def show
+      render json: @reservation
+    end
+
+    def create
+      @reservation = Reservation.new(reservation_params)
+
+      if User.exists?(@reservation.user_id)
+        if @reservation.save
+          render json: @reservation, status: :created
+        else
+          render json: @reservation.errors, status: :unprocessable_entity
+        end
+      else
+        render json: { error: 'User not found' }, status: :not_found
+      end
+    end
+
+    def update
+      if @reservation.update(reservation_params)
+        render json: @reservation
+      else
+        render json: @reservation.errors, status: :unprocessable_entity
+      end
+    end
+
+    def destroy
+      @reservation.destroy
+      head :no_content
+    end
```

Comment:
- [ ] What happens here if the reservation is not successfully destroyed?  Kindly, add some error handling like you did in the update method (line 28). 👍